src/components/files/createThumbnails.ts

```diff
@@ -0,0 +1,77 @@
+// Maps file types (mimetype or ext) to generic
+// thumbnail files
+const genericThumbnails: { [key: string]: string } = {
+  image: 'noun_Image_1570203.png',
+  doc: 'noun_word_3515287.png',
+  docx: 'noun_word_3515287.png',
+  pdf: 'noun_PDF_3283219.png',
+  text: 'noun_word_3515287.png',
+  msword: 'noun_word_3515287.png',
+  file: 'noun_File_3764922.png',
+}
+
+import * as config from '../../config.json'
+import path from 'path'
+import fs from 'fs'
+const sharp = require('sharp')
+const { thumbnailMaxWidth, thumbnailMaxHeight } = config
+
+interface ThumbnailInput {
+  filesPath: string
+  unique_id: string
+  basename: string
+  ext: string
+  mimetype: string
+}
+
+// We use a combination of mimetype (type/subtype) and file extension to
+// determine which conversion process to use, or which generic
+// thumbnail to return
+const createThumbnail = async ({
+  filesPath,
+  unique_id,
+  basename,
+  ext,
+  mimetype,
+}: ThumbnailInput) => {
+  const { type, subtype } = splitMimetype(mimetype)
+
+  const origFilePath = path.join(filesPath, `${basename}_${unique_id}${ext}`)
```

Comment:
I am not sure you've tested this on the latest branch, this should include serial

Reply:
Yeah, my bad. Changed it to save to subfolder and hadn't updated the thumbnail path. Fixed in 24aaeec